### PR TITLE
CI-356: Celebrate page updates

### PR DESCRIPTION
**Pull request by SerhiiBorozenets**
[CI-356](https://cloverpop-internship.atlassian.net/browse/CI-356)

[CI-356]: https://cloverpop-internship.atlassian.net/browse/CI-356?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

CI-458

### CODE REVIEW

**Comment by SerhiiBorozenets** [comment]
Code Climate issues are not relevant

File: app/javascript/components/Pages/modals/CelebrateModal.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  return <Fragment>
+    {show && <div className='backdrop celebrate-modal' /> }
+    <Modal show={show} onHide={() => {setShow(false)}} animation={true} className='modal-dialog-celebrate'>
+      <img src={xClose} className='position-absolute x-close' onClick={() => {setShow(false)}}/>
+      <Modal.Body>
+        <Form>
+          <Form.Group className="mb-3">
+            <div className='fs-5 fw-bold'>
+              Including <span className='red-violet'>@</span>Shoutouts will<br/>allow other team member(s)<br/>to see your response.<br/><br/>
+              <div className='fs-6 muted'>Are you ok with it?</div>
```
Comment (rogergraves) [inline app/javascript/components/Pages/modals/CelebrateModal.js:31]: Please change "Are you ok with it?" to "Is this OK?"

**Review by rogergraves — APPROVED** [review]

File: app/controllers/api/v1/shoutouts_controller.rb [hunk 2]
```diff
@@ -4,15 +4,23 @@ class Api::V1::ShoutoutsController < ApplicationController
...
+    render json: ShoutoutSerializer.new(shoutout).serializable_hash
+  end
+
   def create
-    @shoutout = current_user.shoutouts.new(shoutout_params)
+    @shoutout = if params[:is_celebrate]
+                  current_user.celebrate_shoutouts.new(shoutout_params)
+                else
+                  current_user.shoutouts.new(shoutout_params)
+                end
```
Comment (rogergraves) [inline app/controllers/api/v1/shoutouts_controller.rb:16]: A couple alternative approaches to consider:
```ruby
@shoutout = current_user.celebrate_shoutouts.new(shoutout_params) if params[:is_celebrate]
@shoutout ||= current_user.shoutouts.new(shoutout_params)
```

```ruby
@shoutout = params[:is_celebrate] ? current_user.celebrate_shoutouts.new(shoutout_params) : current_user.shoutouts.new(shoutout_params)
```

**Review by rogergraves — APPROVED** [review]
Please confirm that you looked through CodeClimate issues.

File: app/javascript/components/Pages/modals/CelebrateModal.js (hunk 1 above)
Comment (SerhiiBorozenets) [reply app/javascript/components/Pages/modals/CelebrateModal.js:31]: Thanks, I checked   - Code Climate issues are not relevant.

**Review by SBorozenets — APPROVED** [review]

**Review by SokalMet — APPROVED** [review]

File: app/models/shoutout.rb [hunk 3]
```diff
@@ -30,6 +30,8 @@ class Shoutout < ApplicationRecord
 
   validates :rich_text, presence: true, uniqueness: { scope: %i[user_id time_period_id] }
 
+  scope :not_celebrate, -> { where(type: nil) }
+
   def to_text
     rich_text.gsub(%r{<span class="color-primary">|</span>}, '')
```
Comment (LyubovP) [inline app/models/shoutout.rb:36]: Maybe it's better to use `strip_tags(rich_text)`

Comment (SerhiiBorozenets) [reply app/models/shoutout.rb:36]: @LyubovP this scope needs for separating CelebrateShoutouts from Shoutouts, I am suer that strip_tags(rich_text) can not help with it.

**Review by rogergraves — APPROVED** [review]

**Pull request merged**